sqlserver/tests/test_hc.py

```diff
@@ -0,0 +1,104 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+import time
+from copy import copy
+
+import pytest
+
+from datadog_checks.sqlserver import SQLServer
+
+from .common import CHECK_NAME
+from .utils import HighCardinalityQueries, hc_only
+
+
+@pytest.fixture
+def dbm_instance(instance_docker):
+    instance_docker['dbm'] = True
+    instance_docker['query_metrics'] = {'enabled': True, 'run_sync': True}
+    instance_docker['query_activity'] = {'enabled': True, 'run_sync': True}
+    return copy(instance_docker)
+
+
+@hc_only
+def test_complete_metrics_run(dd_run_check, dbm_instance):
+    dbm_instance['query_activity']['enabled'] = False
+    check = SQLServer(CHECK_NAME, {}, [dbm_instance])
+    queries = HighCardinalityQueries(dbm_instance)
+
+    conn = queries.get_conn_for_user('bob')
+    hc_queries = [queries.create_high_cardinality_query() for _ in range(5000)]
+
+    for q in hc_queries:
+        conn.execute(q)
```

Comment:
How long does this take? We might want to speed this up with parallelization. But no worries if it's only 2s or less.

Also why do you run it before the first check run?